R/data-structure.R

```diff
@@ -5,30 +5,46 @@
 #'
 #' To me, partial matching is often more annoying and surprising than
 #' convenient. It can lead to bugs that are very hard to discover, and I have
-#' been bitten for many times. When I write \code{x$name}, I always mean
-#' precisely \code{name}. You should use a modern code editor to autocomplate
-#' the \code{name} if it is too long to type, instead of using a partial name.
+#' been bitten by it many times. When I write \code{x$name}, I always mean
+#' precisely \code{name}. You should use a modern code editor to autocomplete
+#' the \code{name} if it is too long to type, instead of using partial names.
 #' @param ... Objects (list elements), possibly named. Ignored in the
 #'   \code{print()} method.
 #' @export
-#' @return \code{strict_list()} returns a list with the class
-#'   \code{xfun_strict_list}.
+#' @return \code{strict_list()} and \code{as_strict_list()} both return a list
+#'   with the class \code{xfun_strict_list}. Whereas \code{as_strict_list()}
+#'   attempts to coerce its argument \code{x} to a list if necessary,
+#'   \code{xfun_strict_list} just wraps its argument \code{...} in a list,
+#'   i.e. it will add another list level regardless if \code{...} already is
+#'   of type list.
 #' @examples library(xfun)
-#' (z = strict_list(aaa = 'I am aaa', b = 1:5))
+#' z <- strict_list(aaa = 'I am aaa', b = 1:5)
 #' z$a  # NULL!
 #' z$aaa  # I am aaa
 #' z$b
-#' z$c = 'create a new element'
+#' z$c <-'create a new element'
 #'
-#' z2 = unclass(z)  # a normal list
+#' z2 <- unclass(z)  # a normal list
 #' z2$a  # partial matching
+#'
+#' z3 <- as_strict_list(z2) # a strict list again
+#' z3$a  # NULL again!
 strict_list = function(...) {
   structure(list(...), class = 'xfun_strict_list')
 }
 
 # https://twitter.com/xieyihui/status/782462926862954496
 
-#' @param x A strict list.
+#' @param x For \code{as_strict_list()}, the object to be coerced to a strict
+#'   list.
+#'
+#'   For \code{print()}, a strict list.
+#' @rdname strict_list
+#' @export
+as_strict_list <- function (x) {
```

Comment:
How about `as.strict_list()` instead of `as_strict_list()`?

Reply:
Nooooooo! Please don't 😶 I hate base R for its inconsistency in that regard and do welcome initatives to standardize even "profane" stuff like code style. So I try to adhere to the [tidyverse style guide ](https://style.tidyverse.org/functions.html#naming) whenever possible. Since (almost) all functions use snake case in this package and none the dot-notation (so far), I would definitively stick to that!